### PR TITLE
Esp auto

### PR DESCRIPTION
**Pull request by RadekVana**
Automatically open esp flash on homescreen in case of version mismatch
Remove prusalink from menu
Fix label for undefined IP and MAC

### CODE REVIEW

**Review by vorner — COMMENTED** [review]
There's a bit of minor nitpicking… nothing important.

File: src/common/selftest/selftest_esp_update.hpp [hunk 1]
```diff
@@ -122,19 +152,24 @@ class EspCredentials {
 
     static constexpr const char *file_name = "/usb/prusa_printer_settings.ini";
 
+    static constexpr uint32_t wait_before_wifi_enable_ms = 1024;
```
[the remark this answers is not in the document]
Comment (RadekVana) [reply src/common/selftest/selftest_esp_update.hpp:155]: 1024 is nice round number
512 was not enough 2048 wan unnecessary long

File: src/common/selftest/selftest_esp_update.cpp [hunk 2]
```diff
@@ -336,6 +420,28 @@ void EspCredentials::Loop() {
...
+            // so we wait a bit
+            if ((ticks_ms() - time_stamp) < wait_before_wifi_enable_ms)
+                break;
+            progress_state = esp_credential_action::EnableWIFI;
+            phase = PhasesSelftest::ESP_credentials_enabling_WIFI;
+            time_stamp = ticks_ms();
+            break;
+        case esp_credential_action::EnableWIFI:
+            // at this point cpu load is to high to draw screen nicely
+            // so we wait a bit
```
[the remark this answers is not in the document]
Comment (RadekVana) [reply src/common/selftest/selftest_esp_update.cpp:434]: I needed to add small delay at both places before and after change of GUI state.
Without it it was just very slowly redrawing - especially buttons ad it looked terrible

File: src/common/selftest/selftest_esp_update.cpp [hunk 3]
```diff
@@ -336,6 +420,28 @@ void EspCredentials::Loop() {
...
+            if ((ticks_ms() - time_stamp) < wait_before_wifi_enable_ms)
+                break;
+            netdev_set_active_id(NETDEV_ESP_ID);
```
[the remark this answers is not in the document]
Comment (RadekVana) [reply src/common/selftest/selftest_esp_update.cpp:437]: yes

File: src/common/selftest/selftest_esp_update.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (RadekVana) [reply src/common/selftest/selftest_esp_update.hpp:155]: it does not wait to connect
it just waits until cpu load is lower so screen can be redrawn faster and than it waits
same time until it is redrawn - before calling `netdev_set_active_id(NETDEV_ESP_ID);` 

it was just trials and errors until I got right constant

**Review by vorner — APPROVED** [review]

**Pull request merged**